.github/workflows/docker_build_images.yml

```
@@ -1,22 +1,101 @@
-name: Docker Build Images
+name: Build Docker Images
 on:
+  pull_request:
   push:
     branches:
       - main
     tags:
-      - 'v[2-9][0-9]*.*' # run only on tags greater or equal to v20.0.0 where this new way of building docker image was changed
+      - '*'
 
 concurrency:
-  group: format('{0}-{1}', ${{ github.ref }}, 'Docker Build Images (v20+)')
+  group: format('{0}-{1}', ${{ github.ref }}, 'Docker Build Images')
   cancel-in-progress: true
 
 permissions: read-all
 
 jobs:
+  skip_push:
+    name: Set skip_push if we are on a Pull Request
+    runs-on: ubuntu-20.04
+    if: github.repository == 'vitessio/vitess'
+    outputs:
+      skip_push: ${{ steps.skip_push.outputs.skip_push }}
+
+    steps:
+      - name: Set skip_push
+        id: skip_push
+        run: |
+          push='false'
+          if [[ "${{github.event.pull_request}}" ==  "" ]]; then
+            push='true'
+          fi
+          echo Push='${push}'
+          echo "skip_push=${push}" >> $GITHUB_OUTPUT
+
+  build_and_push_vttestserver:
+    name: Build and push vttestserver
+    runs-on: gh-hosted-runners-16cores-1
+    if: github.repository == 'vitessio/vitess' && needs.skip_push.result == 'success'
+    needs:
+      - skip_push
+
+    strategy:
+      fail-fast: true
+      matrix:
+        branch: [ mysql80 ]
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Login to Docker Hub
+        if: needs.skip_push.outputs.skip_push == 'true'
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Set Dockerfile path
+        run: |
+          echo "DOCKERFILE=./docker/vttestserver/Dockerfile.${{ matrix.branch }}" >> $GITHUB_ENV
+
+      - name: Build and push on main
+        if: startsWith(github.ref, 'refs/tags/') == false
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ${{ env.DOCKERFILE }}
+          push: ${{ needs.skip_push.outputs.skip_push }}
+          tags: vitess/vttestserver:${{ matrix.branch }}
+
+      ######
+      # All code below only applies to new tags
+      ######
+      - name: Get the Git tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: echo "TAG_NAME=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+
+      - name: Set Docker tag name
+        if: startsWith(github.ref, 'refs/tags/')
+        run: |
+          echo "DOCKER_TAG=vitess/vttestserver:${TAG_NAME}-${{ matrix.branch }}" >> $GITHUB_ENV
+
+      - name: Build and push on tags
+        if: startsWith(github.ref, 'refs/tags/')
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ${{ env.DOCKERFILE }}
+          push: true
+          tags: ${{ env.DOCKER_TAG }}
+
+
   build_and_push_lite:
-    name: Build and push vitess/lite Docker images
+    name: Build and push vitess/lite
```

```suggestion
    name: Build and push lite
```

Just to be consistent with vttestserver